Sources/SwiftProtobuf/Enum.swift

```diff
@@ -16,6 +16,7 @@
 // -----------------------------------------------------------------------------
 
 /// Generated enum types conform to this protocol.
+@preconcurrency
```

Comment:
So are these really breaking?

Most of these are internal details and a Developer would never create something that manually conforms to the protocol. So do we really need to annotate them as `@preconcurrency`?

Reply:
@FranzBusch @tbkka @Lukasa

Reply:
These are `public` protocols, so even if it's extremely unlikely someone would go and manually implement types that conform to them, they _could_ do it. This means that public API has changed (because they now conform to `Sendable`) and thus should be annotated with `@preconcurrency`.
I don't think there's any downside to adding these annotations. Have you got any specific concerns?

Reply:
What @gjcairo said here is correct. These have been public protocols and adding the `Sendable` inheritance requires an `@preconcurrency` annotation to downgrade a potential error into a warning when Swift 6 comes around.

Reply:
> they _could_ do it

I don't think we've really historically worried about folks attempting to use these.

> to downgrade a potential error into a warning when Swift 6 comes around

What's the warnings that we aren't seeing yet?

Reply:
Is somebody would conform a type to a protocol that inherits from `Sendable` they will get a warning that their type is required to be `Sendable`. In Swift 6 mode that warning is going to be an error. Adding `@preconcurrency` makes sure that this stays a warning in Swift 6 mode.